### PR TITLE
[Ready] Supercedes #722 and #727 - fixes a number of ansible deployment related issues

### PR DESCRIPTION
**Pull request by rabbah**
This replaces #722 and #727 and completes the fixes for #713.

@domdom82 @lehoanganh @psuter @perryibm @markusthoemmes for your review.

### CODE REVIEW

**Comment by lehoanganh** [comment]
would you add the `hostname: ` by starting zk and kafka?

File: ansible/roles/kafka/tasks/deploy.yml [hunk 1]
```diff
@@ -15,12 +15,17 @@
     ports:
       - "{{ zookeeper.port }}:2181"
 
+# this is important but do not yet know why
+- name: wait before checking if zookeeper is up
+  pause:
+    seconds: 2
```
Comment (lehoanganh) [inline ansible/roles/kafka/tasks/deploy.yml]: haha, me too, still don't know why, if so, why 2, not 3 or 4

Comment (rabbah) [inline ansible/roles/kafka/tasks/deploy.yml]: we will need to dig into the kafka logs and zoookeeper life cycle - likely the isalive check is wrong.

**Comment by rabbah** [comment]
> would you add the hostname: by starting zk and kafka?

are these added everywhere? 
what is the effect of setting this?

**Comment by rabbah** [comment]
Going to address #732 as part of this PR as well.

**Comment by lehoanganh** [comment]
dont know, for this issue not, but if we have a chance to make ansible and ant on-pair, why not do this?

**Comment by psuter** [comment]
I confirm that this solves #713 for me.

**Comment by rabbah** [comment]
Added commit https://github.com/openwhisk/openwhisk/pull/735/commits/57ff94fd1cdf586e3a7b5f701e4f20985fb16060 to fix #732.

**Comment by rabbah** [comment]
@sjfink https://github.com/openwhisk/openwhisk/pull/735/commits/f993aa422834c33f74c1d24e8bbfbfba1b5835f9 for your reviewing pleasure.

@psuter https://github.com/openwhisk/openwhisk/pull/735/commits/57ff94fd1cdf586e3a7b5f701e4f20985fb16060 can you take a look.

**Comment by rabbah** [comment]
PPG 670 approved.

**Comment by lehoanganh** [comment]
LGTM

**Comment by markusthoemmes** [comment]
merged as f73ce9df14366998625ecfe2d1083e4e01f7ccd4 and  abdcb49a2d1a6a043491bf5c7d8955610183fed8

**Pull request closed without merging**